src/unity/python/turicreate/data_structures/sarray.py

```diff
@@ -4399,6 +4399,92 @@ def cumulative_var(self):
         agg_op = "__builtin__cum_var__"
         return SArray(_proxy = self.__proxy__.builtin_cumulative_aggregate(agg_op))
 
+    def filter_by(self, values, exclude=False):
+        
+        """
+        Filter an SArray by values inside an iterable object. Result is an SArray that
+        only includes (or excludes) the values in the given ``values`` :class:`~turicreate.SArray`.
+        If ``values`` is not an SArray, we attempt to convert it to one before filtering.
+        
+        Parameters
+        ----------
+        values : SArray | list | numpy.ndarray | pandas.Series | str
+        The values to use to filter the SArray. The resulting SArray will
+        only include rows that have one of these values in the given
+        column.
+        
+        exclude : bool
+        If True, the result SArray will contain all rows EXCEPT those that
+        have one of the ``values``.
+        
+        Returns
+        -------
+        out : SArray
+        The filtered SArray.
+        
+        Examples
+        --------
+        >>> sa = SArray(['dog', 'cat', 'cow', 'horse'])
+        >>> sa.filter_by(['cat', 'hamster', 'dog', 'fish', 'bird', 'snake'])
+        dtype: str
+        Rows: 2
+        ['dog', 'cat']
+        
+        >>> sa.filter_by(['cat', 'hamster', 'dog', 'fish', 'bird', 'snake'], exclude=True)
+        dtype: str
+        Rows: 2
+        ['horse', 'cow']
+        """
+    
+        from .sframe import SFrame as _SFrame
```

Comment:
Just a style thing -- if this is a local variable, then you don't need the underscore on the _SFrame.  That is just needed at the global level to keep from polluting the module namespace -- making something start with an underscore means it's private to a module or class, so it doesn't show up with tab completion or in our documentation.

Reply:
Actually, probably better to move the import to the global scope and leave the _ on it.

Reply:
Retained the change, as per offline discussion.